File: src/main/java/ru/chernov/urlshortener/utils/StringRandomizer.java
package ru.chernov.urlshortener.utils;

import java.util.Random;
import java.util.concurrent.ThreadLocalRandom;


public class StringRandomizer {
    private static final String UPPER = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    private static final String LOWER = "abcdefghijklmnopqrstuvwxyz";
    private static final String DIGITS = "0123456789";
    private static final String ALPHANUMERIC = UPPER + LOWER + DIGITS;


    public static String nextAlphanumeric(Integer length) {
        Random random = ThreadLocalRandom.current();
        char[] charArray = new char[length];

        for (int i = 0; i < length; i++) {
            int randomIndex = random.nextInt(ALPHANUMERIC.length() - 1);
            charArray[i] = ALPHANUMERIC.charAt(randomIndex);
        }
        return new String(charArray);
    }

}

File: src/main/kotlin/ru/chernov/urlshortener/utils/StringRandomizer.kt
package ru.chernov.urlshortener.utils

import java.util.*
import java.util.concurrent.ThreadLocalRandom


private const val UPPER = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
private const val LOWER = "abcdefghijklmnopqrstuvwxyz"
private const val DIGITS = "0123456789"
private const val ALPHANUMERIC = "$UPPER$LOWER$DIGITS"


fun nextAlphanumeric(length: Int): String {
    val random: Random = ThreadLocalRandom.current()
    val charArray = CharArray(length)

    for (i in 0 until length) {
        val randomIndex: Int = random.nextInt(ALPHANUMERIC.length - 1)
        charArray[i] = ALPHANUMERIC[randomIndex]
    }
    return String(charArray)
}
